src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,29 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
+        }
```

Comment:
The null check for the session seems unnecessary as `factory.openSession()` should either return a valid `Session` object or throw an exception. You can remove the null check to simplify the code.